Atomic/backends/_docker.py

```diff
@@ -537,10 +549,31 @@ def add_string_or_list_to_list(list_item, value):
                                "{}s\n\n removes all containers based on an "
                                "image.".format(iobject.name, iobject.image_name, iobject.name, iobject.name,
                                                iobject.image_name, iobject.image_name, iobject.image_name))
+
+            requested_image = self.has_image(iobject.image) or self.has_image(args.image)
             if iobject.running:
+                if args.replace:
+                    iobject = delete_existing_container(iobject, requested_image, args)
                 return self._running(iobject, args, atomic)
             else:
-                return self._start(iobject, args, atomic)
+                # Container with the name exists
+                image_id = iobject.image
+                # See if we can find an image with the input name
+                if requested_image.id != image_id:
```

Comment:
Ok, I've not had my first cup of tea today, so maybe messing up.  But the comment says we're looking for a match but the if is doing a not match.  Shouldn't this be '==' rather than '!='?